### DISCUSSION TITLE
TextInput readonly prop not working

### DISCUSSION THREAD

**Discussion by reinisg** [carbon-design-system/carbon#13929] [Help] [answered]
Hello!
I am working with React Carbon components
I have packages:
"react": "^17.0.2",
"@carbon/react": "^1.30.0",

Case in code: (sorry can not get example in codesandbox.io because of some @use '@carbon/react'; is not working )
`
 <TextInput id="firstName"
        labelText="First name"
        placeholder="Enter text"
        readOnly={true}
      />
`

Problem: That readOnly prop is not showing Text input as it is in https://react.carbondesignsystem.com/?path=/story/components-textinput--read-only

Can anybody help with this issue?

Regards,
Reinis

**Comment by tw15egan** [comment, marked as answer]
Seems to be working properly with your code:

https://stackblitz.com/edit/github-2moy2y?file=src%2FApp.jsx


Do you possibly have another package that is resolving to an older version of `@carbon/react`?

**Reply by reinisg** [reply]
Ohh, yes!
It was related with some css overrides!

Thank you, for your response and example!
